File: social-hub-api/src/main/java/com/coremedia/blueprint/social/api/MessageContainerDescriptor.java
package com.coremedia.blueprint.social.api;

import com.coremedia.common.annotations.Experimental;
import edu.umd.cs.findbugs.annotations.NonNull;

import java.util.ArrayList;
import java.util.List;

/**
 * A MessageContainerDescriptor configures what message field should be and how when a post container
 * is rendered for an adapter.
 */
@Experimental
public class MessageContainerDescriptor {
  private String propertyName;
  private boolean excluded;
  private String type;
  private Object value;
  private boolean showLabel;

  private List<String> scripts = new ArrayList<>();

  private List<String> scriplets = new ArrayList<>();

  public MessageContainerDescriptor(@NonNull String propertyName, @NonNull MessagePropertyType type) {
    this.propertyName = propertyName;
    this.type = type.name();
  }

  public List<String> getScriplets() {
    return scriplets;
  }

  public List<String> getScripts() {
    return scripts;
  }

  public void addScript(@NonNull String url) {
    this.scripts.add(url);
  }

  public void addScriplet(@NonNull String script) {
    this.scriplets.add(script);
  }

  public boolean isExcluded() {
    return excluded;
  }

  public String getPropertyName() {
    return propertyName;
  }

  public void setExcluded(boolean excluded) {
    this.excluded = excluded;
  }

  public String getType() {
    return type;
  }

  public Object getValue() {
    return value;
  }

  public void setValue(Object value) {
    this.value = value;
  }

  public boolean isShowLabel() {
    return showLabel;
  }

  public void setShowLabel(boolean showLabel) {
    this.showLabel = showLabel;
  }
}

File: apps/studio-client/apps/main/extensions/coremedia-social-hub/src/beans/MessageContainerDescriptor.ts
import SocialHubPropertyNames from "./SocialHubPropertyNames";

class MessageContainerDescriptor {
  #data: any = null;

  constructor(data: any) {
    this.#data = data;
  }

  getValue(): any {
    return this.#data.value;
  }

  getType(): string {
    return this.#data.type;
  }

  showLabel(): boolean {
    return this.#data.showLabel;
  }

  getScripts(): Array<any> {
    return this.#data[SocialHubPropertyNames.DESCRIPTOR_SCRIPTS] || [];
  }

  getScriplets(): Array<any> {
    return this.#data[SocialHubPropertyNames.DESCRIPTOR_SCRIPLETS] || [];
  }

  isExcluded(): boolean {
    return this.#data[SocialHubPropertyNames.DESCRIPTOR_EXCLUDED];
  }

  getPropertyName(): string {
    return this.#data[SocialHubPropertyNames.DESCRIPTOR_PROPERTY_NAME];
  }
}

export default MessageContainerDescriptor;
